src/GoTrueApi.ts

```diff
@@ -326,166 +352,147 @@ export default class GoTrueApi {
+    res.status(200).json({})
   }
 
   /**
-   * Updates the user data.
-   * @param jwt A valid, logged-in JWT.
-   * @param attributes The data you want to update.
+   * Generates links to be sent via email or other.
+   * @param type The link type ("signup" or "magiclink" or "recovery" or "invite").
+   * @param email The user's email.
+   * @param password User password. For signup only.
+   * @param data Optional user metadata. For signup only.
+   * @param redirectTo The link type ("signup" or "magiclink" or "recovery" or "invite").
    */
-  async updateUser(
-    jwt: string,
-    attributes: UserAttributes
-  ): Promise<{ user: User | null; data: User | null; error: ApiError | null }> {
+  async generateLink(
+    type: 'signup' | 'magiclink' | 'recovery' | 'invite',
+    email: string,
+    options: {
+      password?: string
+      data?: object
+      redirectTo?: string
+    } = {}
+  ): Promise<{ data: Session | User | null; error: ApiError | null }> {
     try {
-      const data: any = await put(this.fetch, `${this.url}/user`, attributes, {
-        headers: this._createRequestHeaders(jwt),
-      })
-      return { user: data, data, error: null }
+      const data: any = await post(
+        this.fetch,
+        `${this.url}/admin/generate_link`,
+        {
+          type,
+          email,
+          password: options.password,
+          data: options.data,
+          redirect_to: options.redirectTo,
+        },
+        { headers: this.headers }
+      )
+      return { data, error: null }
     } catch (e) {
-      return { user: null, data: null, error: e as ApiError }
+      return { data: null, error: e as ApiError }
     }
   }
 
+  // User Admin API
+
   /**
-   * Delete a user. Requires a `service_role` key.
+   * Creates a new user.
    *
    * This function should only be called on a server. Never expose your `service_role` key in the browser.
    *
-   * @param uid The user uid you want to remove.
+   * @param attributes The data you want to create the user with.
    * @param jwt A valid JWT. Must be a full-access API key (e.g. service_role key).
    */
-  async deleteUser(
-    uid: string,
-    jwt: string
-  ): Promise<{ user: User | null; data: User | null; error: ApiError | null }> {
+  async createUser(
+    attributes: UserAttributes
+  ): Promise<{ data: null; error: ApiError } | { data: User; error: null }> {
     try {
-      const data: any = await remove(
-        this.fetch,
-        `${this.url}/admin/users/${uid}`,
-        {},
-        {
-          headers: this._createRequestHeaders(jwt),
-        }
-      )
-      return { user: data, data, error: null }
+      const data: any = await post(this.fetch, `${this.url}/admin/users`, attributes, {
+        headers: this.headers,
+      })
+      return { data, error: null }
     } catch (e) {
-      return { user: null, data: null, error: e as ApiError }
+      return { data: null, error: e as ApiError }
```

Comment:
will removing the user key (even though it returns null) result in a breaking change?